### DISCUSSION TITLE
not-required with regex?

### DISCUSSION THREAD

**Discussion by girish-tharwani** [json-schema-org/community#618] [Q&A] [answered]
Hello Experts
I understand that a schema like this will help me forbid "foo" and "bar" in my json but is there a way to generalize it to block the case variants also (e.g. FOO, Foo, BAR, Bar etc)

```
{
    "not": {
        "anyOf": [
            {"required": ["foo"]},
            {"required": ["bar"]},
        ]
    }
}

```

**Comment by jdesrosiers** [comment, marked as answer]
There aren't good options for working with case-insensitivity in JSON Schema. Any solution is going to be ugly. This is the best I've got.

```json
"propertyNames": {
  "not": {
     "anyOf": [
        { "pattern": "^[fF][oO][oO]$" },
        { "pattern": "^[bB][aA][rR]$" }
    ]
  }
}
```
